assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -155,19 +155,33 @@ function MediaProvider({ children }) {
 
   useEffect(generatePoster, [media, mediaType, searchTerm]);
 
-  const context = {
-    state: { ...state, isUploading },
-    actions: {
+  const context = useMemo(
```

Comment:
Are we seeing performance drain when we load image files for display or when we fetch "lists of images"? I.e. my first question here is do we re-enter this provider at all.

Reply:
Loading a page of images MediaProvider causes MediaProvider to be re-rendered, but it doesn't appear to be a performance drain. I think useMemo is unnecessary here.

Reply:
Makes sense. Generally looking at this provider I'm still not quite happy with it structurally: it has things like `pageNum`, but that doesn't seem like a right level of detail for the global state.